modules/strelka/1.0/strelka.smk

```diff
@@ -0,0 +1,213 @@
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["strelka"]`
+CFG = op.setup_module(
+    name = "strelka",
+    version = "1.0",
+    subdirectories = ["inputs", "strelka", "filtered", "outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _strelka_input_bam,
+    _strelka_input_vcf,
+    _strelka_configure_paired,
+    _strelka_configure_unpaired,
+    _strelka_unzip,
+    _strelka_filter,
+    _strelka_output_vcf,
+    _strelka_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _strelka_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+
+
+rule _strelka_input_vcf:
+    input:
+        vcf = CFG["inputs"].get("candidate_indel")
+    output:
+        vcf = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/vcf/{tumour_id}--{normal_id}--{pair_status}.candidateSmallIndels.vcf.gz"
+    conda:
+        CFG["conda_envs"]["tabix"]
+    shell:
+        "bgzip -c {input.vcf} > {output.vcf} && tabix -p vcf {output.vcf}"
+
+
+# Example variant calling rule (multi-threaded; must be run on compute server/cluster)
+rule _strelka_configure_paired: # Somatic
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{tumour_id}.bam",
+        normal_bam = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{normal_id}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        indels = rules._strelka_input_vcf.output.vcf
+    output:
+        runwf = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/runWorkflow.py"
+    log:
+        stdout = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_configure.stdout.log",
+        stderr = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_configure.stderr.log"
+    params:
+        opts = op.switch_on_wildcard("seq_type", CFG["options"]["configure"]),
+    wildcard_constraints:
+        pair_status = "matched|unmatched"
+    conda:
+        CFG["conda_envs"]["strelka"]
+    shell:
+        op.as_one_line("""
+        configureStrelkaSomaticWorkflow.py 
+        --normalBam={input.normal_bam}
+        --tumorBam={input.tumour_bam}
+        --referenceFasta={input.fasta}
+        --runDir=$(dirname {output.runwf})
+        --indelCandidates={input.indels}
+        {params.opts} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+rule _strelka_configure_unpaired: # germline
```

Comment:
I'm afraid that users will accidentally run tumours in `no_normal` mode and expect a somatic variant calling model to be used (rather than the germline model used here). Maybe add a message directive to this rule warning the user about this. 

https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#messages